### PR TITLE
RDF/XML parser

### PR DESCRIPTION
**Pull request by BertCommeine**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -114,5 +115,8 @@
       "dependencyPin": "Pin ${dependency} to ${oldVersion}",
       "devDependencyPin": "Pin ${dependency} to ${oldVersion}"
     }
+  },
+  "dependencies": {
+    "rdfxml-streaming-parser": "^1.0.0"
```
Comment (rubensworks) [inline package.json]: This dependency should not be added on the root, only in `packages/actor-rdf-parser-rdfxml`.

File: packages/actor-init-rdf-parse/config/config-example.json [hunk 2]
```diff
@@ -31,6 +31,11 @@
           "@id": "config-example:myRdfParserJsonLd",
           "@type": "ActorRdfParseJsonLd",
           "priorityScale": 0.5
+        },
+        {
+          "@id": "config-example:myRdfParserRdfXml",
```
Comment (rubensworks) [inline packages/actor-init-rdf-parse/config/config-example.json]: You'll have to add the following line at the top of this file:

```
"https://linkedsoftwaredependencies.org/bundles/npm/@comunica/actor-rdf-parse-rdfxml/^1.0.0/components/context.jsonld",
```

File: packages/actor-rdf-parse-rdfxml/components/Actor/RdfParse/RdfXml.jsonld [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  "@id": "npmd:@comunica/actor-rdf-parse-rdfxml",
+  "components": [
+    {
+      "@id": "carpr:Actor/RdfParse/RdfXml",
+      "@type": "Class",
+      "extends": "cbrp:Actor/RdfParseFixedMediaTypes",
+      "requireElement": "ActorRdfParseRdfXml",
+      "comment": "A comunica RDF/XML RDF Parse Actor.",
+      "parameters": [
+              {
```
Comment (rubensworks) [inline packages/actor-rdf-parse-rdfxml/components/Actor/RdfParse/RdfXml.jsonld]: The indentation seems to be too deep here.

File: packages/actor-rdf-parse-rdfxml/lib/ActorRdfParseRdfXml.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    super(args);
+  }
+
+  public async runHandle(action: IActionRdfParse, mediaType: string, context: ActionContext)
+    : Promise<IActorRdfParseOutput> {
+    action.input.on('error', (e) => quads.emit('error', e));
+    const quads = action.input.pipe(new RdfXmlParser());
+    return {
+      quads,
+      triples: mediaType === 'application/rdf+xml'};
```
Comment (rubensworks) [inline packages/actor-rdf-parse-rdfxml/lib/ActorRdfParseRdfXml.ts]: `triples` is always true for RDF/XML, so just set `triples: true`, and you can ignore the `mediaType`.
Also, put the `}` on a new line.

File: packages/actor-rdf-parse-rdfxml/test/ActorRdfParseRdfXml-test.ts [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+  });
+
+  describe('An ActorRdfParseRdfXml instance', () => {
+    let actor: ActorRdfParseRdfXml;
+    let input: Readable;
+    let inputError: Readable;
+
+    beforeEach(() => {
+      actor = new ActorRdfParseRdfXml({ bus, mediaTypes: {
+        'application/trig': 1.0,
```
Comment (rubensworks) [inline packages/actor-rdf-parse-rdfxml/test/ActorRdfParseRdfXml-test.ts]: For consistency, just put one mediatype (the RDF/XML one) in this object.

File: packages/actor-rdf-parse-rdfxml/test/ActorRdfParseRdfXml-test.ts [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+    });
+
+    describe('for getting media types', () => {
+      it('should test', () => {
+        return expect(actor.test({ mediaTypes: true })).resolves.toBeTruthy();
+      });
+
+      it('should run', () => {
+        return expect(actor.run({ mediaTypes: true })).resolves.toEqual({ mediaTypes: {
+          'application/trig': 1.0,
```
Comment (rubensworks) [inline packages/actor-rdf-parse-rdfxml/test/ActorRdfParseRdfXml-test.ts]: Also just one RDF/XML mediatype here.

**Comment by rubensworks** [comment]
@BertCommeine After the build has been fixed, be sure to sign the CLA (see first comment).

**Comment by RubenVerborgh** [comment]
Thanks guys.

Note for the future: we definitely need to think about code splitting and on-demand loading, as having many parsers on the client side significantly increases size. We probably have an issue for that.

**Comment by rubensworks** [comment]
Indeed, the issue has been open for a while: #9

We should perhaps include this on our roadmap at some point.

**Pull request merged**